10_rest/alextmz/pkg/rest/rest.go

```diff
@@ -0,0 +1,122 @@
+package rest
+
+import (
+	"net/http"
+	"path"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy"
+	"github.com/go-chi/render"
+)
+
+type HTTPHandler struct {
+	Store puppy.Storer
+}
+
+func (ht HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		ht.handleGet(w, r)
+	case "PUT":
+		ht.handlePut(w, r)
+	case "POST":
+		ht.handlePost(w, r)
+	case "DELETE":
+		ht.handleDelete(w, r)
+	}
+}
+
+func (ht *HTTPHandler) handleGet(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(path.Base(r.RequestURI))
+	if isError(w, err) {
+		return
+	}
+
+	puppy, err := ht.Store.ReadPuppy(id)
+
```

Comment:
sometimes you've got new lines here, sometimes you don't have them. This is NIT and can be easily ignored